include/bluetooth/l2cap.h

```diff
@@ -26,21 +26,61 @@
 extern "C" {
 #endif
 
-/** L2CAP header size, used for buffer size calculations */
+/** L2CAP PDU header size, used for buffer size calculations */
 #define BT_L2CAP_HDR_SIZE               4
 
+/** Maximum Transmission Unit (MTU) for an outgoing L2CAP PDU. */
+#define BT_L2CAP_TX_MTU (CONFIG_BT_L2CAP_TX_MTU)
+
+/** Maximum Transmission Unit (MTU) for an incoming L2CAP PDU. */
+#define BT_L2CAP_RX_MTU (CONFIG_BT_BUF_ACL_RX_SIZE - BT_L2CAP_HDR_SIZE)
+
 /** @def BT_L2CAP_BUF_SIZE
  *
- *  @brief Helper to calculate needed outgoing buffer size, useful e.g. for
- *  creating buffer pools.
+ *  @brief Helper to calculate needed buffer size for L2CAP PDUs.
+ *         Useful for creating buffer pools.
+ *
+ *  @param mtu Needed L2CAP PDU MTU.
+ *
+ *  @return Needed buffer size to match the requested L2CAP PDU MTU.
+ */
+#define BT_L2CAP_BUF_SIZE(mtu) BT_BUF_ACL_SIZE(BT_L2CAP_HDR_SIZE + (mtu))
+
+/** L2CAP SDU header size, used for buffer size calculations */
+#define BT_L2CAP_SDU_HDR_SIZE           2
+
+/** @brief Maximum Transmission Unit for an unsegmented outgoing L2CAP SDU.
+ *
+ *  The Maximum Transmission Unit for an outgoing L2CAP SDU when sent without
+ *  segmentation, i.e a single L2CAP SDU will fit inside a single L2CAP PDU.
+ *
+ *  The MTU for outgoing L2CAP SDUs with segmentation is defined by the
+ *  size of the application buffer pool.
+ */
+#define BT_L2CAP_SDU_TX_MTU (BT_L2CAP_TX_MTU - BT_L2CAP_SDU_HDR_SIZE)
+
+/** @brief Maximum Transmission Unit for an unsegmented incoming L2CAP SDU.
  *
- *  @param mtu Needed L2CAP MTU.
+ *  The Maximum Transmission Unit for an incoming L2CAP SDU when sent without
+ *  segmentation, i.e a single L2CAP SDU will fit inside a single L2CAP PDU.
  *
- *  @return Needed buffer size to match the requested L2CAP MTU.
+ *  The MTU for incoming L2CAP SDUs with segmentation is defined by the
+ *  size of the application buffer pool. The application will have to define
+ *  define an alloc_buf callback for the channel in order to support receiving
+ *  segmented L2CAP SDUs.
  */
-#define BT_L2CAP_BUF_SIZE(mtu) (BT_BUF_RESERVE + \
-				BT_HCI_ACL_HDR_SIZE + BT_L2CAP_HDR_SIZE + \
-				(mtu))
+#define BT_L2CAP_SDU_RX_MTU (BT_L2CAP_RX_MTU - BT_L2CAP_SDU_HDR_SIZE)
+
+/** @def BT_L2CAP_SDU_BUF_SIZE
+ *
+ *  @brief Helper to calculate needed buffer size for L2CAP SDUs.
+ *         Useful for creating buffer pools.
+ *
+ *  @param mtu Needed L2CAP SDU MTU.
```

Comment:
Sounds a little repetitive to use SDU MTU, in fact why not just use BT_L2CAP_*_SDU.